production/migrations/20230621162215_courses.sql

```diff
@@ -96,14 +98,40 @@ SELECT
                 c.id = co.course_id AND
                 co.year = (SELECT EXTRACT(YEAR FROM CURRENT_DATE) AS current_year)
         LIMIT 1
-    ) AS offerings
+    ) AS offerings,
+    COALESCE(
+        NULLIF(array_agg(DISTINCT rc.subject_code || rc.catalog_number::TEXT), '{null}'),
+        '{}'
+    ) AS required_prerequisites,
+    COALESCE(
+        NULLIF(array_agg(DISTINCT oc.subject_code || oc.catalog_number::TEXT), '{null}'),
+        '{}'
+    ) AS optional_prerequisites
 FROM
     courses c
-        LEFT JOIN course_offerings co ON c.id = co.course_id
+    LEFT JOIN course_offerings co ON c.id = co.course_id
+    LEFT JOIN prerequisites p ON c.prerequisites_id = p.id
+    LEFT JOIN required_prerequisites rp ON p.id = rp.prerequisite_id
+    LEFT JOIN courses rc ON rp.course_id = rc.id
+    LEFT JOIN optional_prerequisites op ON p.id = op.prerequisite_id
+    LEFT JOIN courses oc ON op.course_id = oc.id
 GROUP BY
     c.id,
     c.subject_code
 ORDER BY
     c.catalog_number;
 
+-- Courses Courses MV Indices
+CREATE UNIQUE INDEX IF NOT EXISTS idx_mv_courses_subject_code_catalog_number
```

Comment:
Add indexes to MV (which is persisted on disk) to increase scan lookup performance similar to base tables.